src/common/config.cpp

```diff
@@ -525,21 +535,16 @@ void load(const std::filesystem::path& path) {
         m_window_size_W = toml::find_or<int>(gui, "mw_width", 0);
         m_window_size_H = toml::find_or<int>(gui, "mw_height", 0);
 
-        auto old_game_install_dir = toml::find_fs_path_or(gui, "installDir", {});
-        if (!old_game_install_dir.empty()) {
-            settings_install_dirs.push_back(old_game_install_dir);
-            data.as_table().erase("installDir");
-        }
-
         const auto install_dir_array =
             toml::find_or<std::vector<std::string>>(gui, "installDirs", {});
         for (const auto& dir : install_dir_array) {
-            bool not_already_included =
-                std::find(settings_install_dirs.begin(), settings_install_dirs.end(), dir) ==
-                settings_install_dirs.end();
-            if (not_already_included) {
-                settings_install_dirs.emplace_back(std::filesystem::path{dir});
-            }
+            settings_install_dirs.emplace_back(std::filesystem::path{dir});
+        }
+
+        // TODO Migration code, after a major release this should be removed.
+        auto old_game_install_dir = toml::find_fs_path_or(gui, "installDir", {});
+        if (!old_game_install_dir.empty()) {
+            addGameInstallDir(old_game_install_dir);
```

Comment:
I would suggest moving this above and use an else condition to switch to the installDirs. Also, we don't need use `addGameInstallDir(old_game_install_dir);` rather just push the string to settings_install_dirs.
```suggestion
        // TODO Migration code, after a major release this should be removed.
        auto old_game_install_dir = toml::find_fs_path_or(gui, "installDir", {});
        if (!old_game_install_dir.empty()) {
            addGameInstallDir(old_game_install_dir);
```

Reply:
https://github.com/shadps4-emu/shadPS4/pull/1329/commits/2144b00de539490957d9e7437e5e2136bc5c267b